RT_SYCL/vec.hpp

```diff
@@ -83,7 +83,7 @@ vec random_in_unit_disk()
 // Computes refracted ray's direction based on refractive index
 vec refract(const vec& uv, const vec& n, float etai_over_etat)
 {
-    auto cos_theta = sycl::fmin(sycl::dot(-uv, n), 1.0f);
+    auto cos_theta = sycl::fmin(sycl::dot(vec{}-uv, n), 1.0f);
```

Comment:
This is more efficient anyway:
```suggestion
    auto cos_theta = sycl::fmin(-sycl::dot(uv, n), 1.0f);
```